Project updates

friederbluemle opened the pull request:
A bunch of project cleanups and dependency updates. Best reviewed commit-by-commit.

Let me know if you prefer two separate PRs for cleanup and dependencies.

<sub><sup>Please NO GitHub rebase/squash merges. If you'd like me to combine some or all of the commits, please let me know, and I will update my branch. Thank you.</sup></sub>

luizgrp commented on README.md:
```diff
@@ -17,27 +18,28 @@ In addition, each Section can have its state(Loading/Loaded/Failed) controlled i
     @Override
     public int getContentItemsTotal() {
-        return myList.size(); // number of items of this section
+        return mItemList.size(); // number of items of this section
```
I'd rather not use Hungarian notation! same for tvItem below

friederbluemle replied:
Even though the `m` prefix looks very similar to Hungarian notation, it actually identifies the *scope*, not the *type* of the variable, as defined in the [Field Naming Conventions](https://source.android.com/source/code-style#follow-field-naming-conventions) of the AOSP style guide.
But if you prefer variable names without the prefix, that's fine too ;) I removed the prefixes in my updated PR.

luizgrp replied:
Thanks for the link and explanation! But my preferences are more aligned with the reasons of this [link](http://jakewharton.com/just-say-no-to-hungarian-notation/). Thanks for updating it!

luizgrp commented:
Hi @friederbluemle, many thanks for your PR!
Once you address the comment above I will get it merged 👍

friederbluemle commented:
@luizgrp Thanks for the review. I updated my PR, and rebased my branch to the latest `master`. Is `master` the correct branch to target, or should it be `develop`?

luizgrp commented:
It should be develop for now on, but I only created it yesterday and forgot to tell you, so I will merge this one into master no problem. I still have to write a "contributing" guide.
Thanks again!
